tests/unit/test_common.py

```diff
@@ -71,6 +72,59 @@ def test_read_from_contentstore(self):
         # Assert
         self.assertEqual(content, "bogus")
 
+    def test_read_from_contentstore_retry_succeed(self):
+        # Setup
+        asset_mock = Mock()
+        asset_mock.data = "bogus"
+        contentstore_mock = Mock()
+        contentstore_mock.return_value.find.side_effect = [
+            PyMongoError,
+            PyMongoError,
+            asset_mock
+        ]
+        django_mock = Mock()
+        django_mock.contentstore = contentstore_mock
+
+        # Run
+        with patch.dict('sys.modules', **{
+            'xmodule': Mock(),
+            'xmodule.contentstore': Mock(),
+            'xmodule.contentstore.django': django_mock,
+            'xmodule.contentstore.content': Mock(),
+            'opaque_keys': Mock(),
+            'opaque_keys.edx': Mock(),
+            'opaque_keys.edx.locator': Mock(),
+        }):
+            content = read_from_contentstore("course_id", "path")
+
+        # Assert
+        self.assertEqual(content, "bogus")
+        self.assertEqual(contentstore_mock.return_value.find.call_count, 3)
+
+    def test_read_from_contentstore_retry_fail(self):
+        # Setup
+        contentstore_mock = Mock()
+        contentstore_mock.return_value.find.side_effect = [
+            PyMongoError,
```

Comment:
Small suggestion: when code catches the abstract superclass of an exception category (in our case, `PyMongoError`), then I think it's better to write the side effects in the test code so that they throw concrete subclasses.

I've pushed a fixup commit doing that in 19f12a7924135f3fe40edbd395b5c149cf3c44ea, please see if that makes sense to you.

Reply:
OK, I've squashed the fixup commit and force-pushed to your branch. I think this is good to merge, then.

Reply:
Ack, thanks!